src/components/Button/button.test.tsx

```diff
@@ -163,22 +163,42 @@ describe('<Button />', () => {
     });
   });
 
-  describe('when clicked', () => {
+  describe('with mouse events', () => {
     let clicked: boolean;
-    function clickMe() {
+    let hoveredIn: boolean;
+    let hoveredOut: boolean;
+    function onClick() {
       clicked = true;
     }
+    function onHoverIn() {
+      hoveredIn = true;
+    }
+    function onHoverOut() {
+      hoveredOut = true;
+    }
     beforeEach(() => {
       clicked = false;  
+      hoveredIn = false;  
+      hoveredOut = false;
       component = shallow(
-        <Button text={sampleText} onClick={clickMe} />,
+        <Button text={sampleText} onClick={onClick} onHoverIn={onHoverIn} onHoverOut={onHoverOut} />,
       );
     });
 
     it('triggers its onClick callback', () => {
       component.simulate('click');
       expect(clicked).toBe(true);
```

Comment:
In combination with the feedback from @unindented  (spies), recommend leveraging `describe` to describe scenarios and `it` to specify expectations:

```javascript
describe('when the component is clicked', function() {
  beforeEach(function() {
    component.simulate('click');
  });

  it('triggers the onClick callback', function() {
    expect(onClick).toHaveBeenCalled();
  });
});